src/protocol/mod.rs

```diff
@@ -222,11 +222,22 @@ impl Protocol {
         })
     }
 
-    pub fn load(executable_path: &Path) -> R<Vec<Protocol>> {
-        let file_contents = read_protocols_file(executable_path)?;
+    fn parse_protocols(file_contents: &str) -> R<Vec<Protocol>> {
         let yaml: Vec<Yaml> = YamlLoader::load_from_str(&file_contents)?;
         yaml.into_iter().map(Protocol::parse).collect()
     }
+
+    pub fn load(executable_path: &Path) -> R<Vec<Protocol>> {
+        let file_contents = read_protocols_file(executable_path)?;
+        let result = Protocol::parse_protocols(&file_contents).map_err(|error| {
+            format!(
+                "error parsing {}: {}",
```

Comment:
i think if the message mentions YAML specifically it would be more clear, something like: `YAML parsing error in: {}`

Reply:
The file name will always include the `.yaml` extension, that's why I thought it wouldn't be necessary. Do you still think this change would be useful?